test/unit/test_base_service.rb

```diff
@@ -184,4 +191,104 @@ def test_dummy_request_icp
     service_response = service.request(method: "GET", url: "/music", headers: {})
     assert_equal(response, service_response.result)
   end
+
+  def test_configure_service_using_credentials_in_env
+    file_path = File.join(File.dirname(__FILE__), "../../resources/ibm-credentials.env")
+    authenticator = IBMCloudSdkCore::BasicAuthenticator.new(
+      username: "apikey",
+      password: "icp-xyz"
+    )
+    service = IBMCloudSdkCore::BaseService.new(
+      authenticator: authenticator,
+      service_name: "visual_recognition"
+    )
+    assert_nil(service.service_url)
+    assert_equal(false, service.disable_ssl_verification)
+    ENV["IBM_CREDENTIALS_FILE"] = file_path
+    service.configure_service("visual_recognition")
+
+    assert_equal("https://gateway.ronaldo.com", service.service_url)
+    assert(service.disable_ssl_verification)
+    ENV.delete("IBM_CREDENTIALS_FILE")
+  end
+
+  def test_configure_service_using_credentials_in_vcap_match_outer_key
+    authenticator = IBMCloudSdkCore::BasicAuthenticator.new(
+      username: "apikey",
+      password: "icp-xyz"
+    )
+    service = IBMCloudSdkCore::BaseService.new(
+      authenticator: authenticator,
+      service_name: "key_to_service_entry_1"
+    )
+    ENV["VCAP_SERVICES"] = JSON.parse(File.read(Dir.getwd + "/resources/vcap-testing.json")).to_json
+    service.configure_service("key_to_service_entry_1")
+
+    assert_equal("https://on.the.toolchainplatform.net/devops-insights/api", service.service_url)
+    ENV.delete("VCAP_SERVICES")
+  end
+
+  def test_configure_service_using_credentials_in_vcap_match_inner_key
+    authenticator = IBMCloudSdkCore::BasicAuthenticator.new(
+      username: "apikey",
+      password: "icp-xyz"
+    )
+    service = IBMCloudSdkCore::BaseService.new(
+      authenticator: authenticator,
+      service_name: "devops_insights"
```

Comment:
For this test (matching a service entry's "name" field), let's change the test so that it matches on an entry that is not the first one in the list.   (Yes, I'm paranoid).

Reply:
Looks like I was ahead of you on this one. The service with the name "devops_insights" actually belongs to the second item in that list